service/src/global_peer_updater.rs

```diff
@@ -0,0 +1,64 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+	Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{
+	error::Error,
+	globals::worker::GetMutWorker,
+	worker::{UpdatePeers, WorkerResult},
+};
+use log::*;
+use std::sync::Arc;
+
+#[cfg(test)]
+use mockall::predicate::*;
+#[cfg(test)]
+use mockall::*;
+
+/// Updates the peers of the global worker.
+#[cfg_attr(test, automock)]
+pub trait UpdateWorkerPeers {
+	fn update_peers(&self) -> WorkerResult<()>;
+}
+
+pub struct GlobalPeerUpdater<Worker> {
+	worker: Arc<Worker>,
+}
+
+impl<Worker> GlobalPeerUpdater<Worker> {
+	pub fn new(worker: Arc<Worker>) -> Self {
+		GlobalPeerUpdater { worker }
+	}
+}
+
+// FIXME: We should write unit tests for this one here - but with the global worker struct, which is not yet made to be mocked,
```

Comment:
Tried it, but that would introduce many more changes in this PR, so I left it at this comment

Reply:
I think this is too granular. I can agree to an extra trait, but another global worker something instance is an overkill without any benefit in my opinion.

Reply:
What's your suggestion to access the global worker then? 
WIth the newly introduced peers variable, that need to be updated from time to time, we need to access that global struct. Ofc, it's arguable if the peers need to be stored as a variable. But I think they should be stored somewhere, probably not as url but as a ws client to keep the connection open. And the worker struct seemed to be the right place for that.

Reply:
Felix introduces a `PeerFetcher` struct here : https://github.com/integritee-network/worker/pull/580/files#diff-02cc9e60074a999adfbd489c2cd472bbf1023af88b8d1ff367c6d378f445cfd8R38

Should I follow that example, and include the peer storage right there and not in the global worker? We will need to pass a PeerFetcher struct as a parameter instead when constructing the global worker, so the worker can call a "fetch_peer" or something, but we do not need to change the worker itself on a regular basis.

Can you agree with that approach?

Reply:
This is something where Felix still couldn't convince me. Why is it better to have:

```
let w1 = Arc::new(Worker::new());
let fetcher = `GlobalPeerFetcher::new(worker.clone())`;
let gossiper = `GlobalGossiper::new(worker.clone())`;
lef updateWorker = `GlobalUpdater::(worker.clone());
```

instead of implementing the traits for the single worker, we have? There is only one worker to begin with that has countless reference pointers. I believe this is a pattern that does just not apply to rust.

This is so much boilerplate code, which I don't understand why we have it, but now that it seems to be the flow, I won't stand in the way. It is better than having a hybrid flow.

Reply:
The construct itself stand in its way, because it makes unit testing impossible as it is now. So I agree with you here - I'd prefer to have traits for a single worker, it would make things less complicated. But I don't think this PR is the right place to correct that. 

I think with the PeerFetcher approach suggested above we introduce relatively few changes and it won't matter much if the whole worker struct is changed later on, as it's quite a standalone struct / trait.
What do you think?

Reply:
Yes, as I said above. I can live with how it is.

Reply:
Okay, leaving it as it is now. The future will show where we go.